java-util/src/main/java/io/druid/java/util/common/granularity/PeriodGranularity.java

```diff
@@ -352,56 +430,29 @@ private long truncateMillisPeriod(final long t)
   }
 
   @Override
-  public byte[] cacheKey()
+  public void serialize(
+      JsonGenerator jsonGenerator, SerializerProvider serializerProvider
+  ) throws IOException, JsonProcessingException
   {
-    return StringUtils.toUtf8(period.toString() + ":" + chronology.getZone().toString() + ":" + origin);
-  }
-
-  @Override
-  public boolean equals(Object o)
-  {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-
-    PeriodGranularity that = (PeriodGranularity) o;
-
-    if (hasOrigin != that.hasOrigin) {
-      return false;
-    }
-    if (origin != that.origin) {
-      return false;
+    // Retain the same behavior as pre-refactor granularity code.
```

Comment:
Once this code is accepted, "pre-refactor granularity code" isn't going to make much sense to people.  Instead, explain that an old API accepted only the String representation so this refactor required us to maintain that when serializing the object for compatibility purposes.

It's likely also possible that this can be removed in a future release after this has been accepted and is running, right?